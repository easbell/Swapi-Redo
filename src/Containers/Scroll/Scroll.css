.fade {
  position: relative;
  width: 100%;
  max-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  overflow: hidden;
  display: flex;
  flex-direction: row;
  justify-content: center;
  position: relative;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 140%;
  perspective: 400px;
  text-align: justify;
}

.crawl {
  position: relative;
  transform-origin: 20% 100%;
  animation: crawl 450s linear;
}

@keyframes crawl {
	0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
	100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.enter-app {
  border-radius: 5px;
  font-size: 18px;
  padding: 9px;
  margin: 5px 33px;
  padding-top: 36px;
  background-color: #545558;
  color: white;
  border: 0;
  width: 1800px;
  height: 107px;
  text-decoration: none;
  position: relative;
  top: 50px;
  left: 50px;
}

.enter-app:hover {
  background-color: #fff;
  color: #545558;
  cursor: pointer;
}