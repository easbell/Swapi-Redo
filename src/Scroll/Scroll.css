.fade {
  position: relative;
  width: 100%;
  max-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  overflow: hidden;
  display: flex;
  justify-content: center;
  position: relative;
  height: 300px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 200px;
  text-align: justify;
}

.crawl {
  position: relative;
  transform-origin: 20% 100%;
  animation: crawl 450s linear;
}

@keyframes crawl {
	0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
	100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}