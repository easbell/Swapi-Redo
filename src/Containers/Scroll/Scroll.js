import React from 'react';
import './Scroll.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const Scroll = (props) => {
  return (
    <div className='fade'>
      <div className='star-wars'>
      <Link to={'/main'} className='enter-app'>Enter App</Link>
        <div className='crawl'>
          <p>{props.film.openingCrawl}</p>
          <p>{props.film.title}</p>
          <p>{props.film.releaseDate}</p>
        </div>
      </div>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  film: state.film
})

export default connect(mapStateToProps)(Scroll);