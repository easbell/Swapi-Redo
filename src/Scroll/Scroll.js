import React from 'react';
import './Scroll.css';
import { connect } from 'react-redux';

export const Scroll = (props) => {
  const handleClick = (e) => {
    e.preventDefault();
    props.enterApp();
  }

  return (
    <div className='fade'>
      <div className='star-wars'>
      <button onClick={handleClick} className="enter-app">Enter App</button>
        <div className='crawl'>
          <p>{props.film.openingCrawl}</p>
          <p>{props.film.title}</p>
          <p>{props.film.releaseDate}</p>
        </div>
      </div>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  film: state.film
})

export default connect(mapStateToProps)(Scroll);