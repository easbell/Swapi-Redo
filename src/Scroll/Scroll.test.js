import React from 'react';
import { Scroll } from './Scroll';
import { shallow } from 'enzyme';

describe('Scroll', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      film: {openingCrawl: "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
      releaseDate: "2005-05-19",
      title: "Revenge of the Sith"}
    }

    wrapper = shallow(<Scroll film={props.film}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire enterApp when button is clicked', () => {
    const mockFn = jest.fn();
    wrapper = shallow(<Scroll film={props.film} enterApp={mockFn}/>)
    const mockedEvent = { target: {}, preventDefault: () => {}}
    wrapper.find('.enter-app').simulate('click', mockedEvent)

    expect(mockFn).toHaveBeenCalled();
  });
})
