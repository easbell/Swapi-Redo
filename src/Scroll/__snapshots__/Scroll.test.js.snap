// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll
    film={
      Object {
        "openingCrawl": "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
        "releaseDate": "2005-05-19",
        "title": "Revenge of the Sith",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="star-wars"
      >
        <button
          className="enter-app"
          onClick={[Function]}
        >
          Enter App
        </button>
        <div
          className="crawl"
        >
          <p>
            War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
          </p>
          <p>
            Revenge of the Sith
          </p>
          <p>
            2005-05-19
          </p>
        </div>
      </div>,
      "className": "fade",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="enter-app"
            onClick={[Function]}
          >
            Enter App
          </button>,
          <div
            className="crawl"
          >
            <p>
              War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
            </p>
            <p>
              Revenge of the Sith
            </p>
            <p>
              2005-05-19
            </p>
          </div>,
        ],
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter App",
            "className": "enter-app",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Enter App",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
              </p>,
              <p>
                Revenge of the Sith
              </p>,
              <p>
                2005-05-19
              </p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
              },
              "ref": null,
              "rendered": "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Revenge of the Sith",
              },
              "ref": null,
              "rendered": "Revenge of the Sith",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2005-05-19",
              },
              "ref": null,
              "rendered": "2005-05-19",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="star-wars"
        >
          <button
            className="enter-app"
            onClick={[Function]}
          >
            Enter App
          </button>
          <div
            className="crawl"
          >
            <p>
              War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
            </p>
            <p>
              Revenge of the Sith
            </p>
            <p>
              2005-05-19
            </p>
          </div>
        </div>,
        "className": "fade",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="enter-app"
              onClick={[Function]}
            >
              Enter App
            </button>,
            <div
              className="crawl"
            >
              <p>
                War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
              </p>
              <p>
                Revenge of the Sith
              </p>
              <p>
                2005-05-19
              </p>
            </div>,
          ],
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter App",
              "className": "enter-app",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Enter App",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....
                </p>,
                <p>
                  Revenge of the Sith
                </p>,
                <p>
                  2005-05-19
                </p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
                },
                "ref": null,
                "rendered": "War! The Republic is crumblingunder attacks by the ruthlessSith Lord, Count Dooku.There are heroes on both sides.Evil is everywhere.In a stunning move, thefiendish droid leader, GeneralGrievous, has swept into theRepublic capital and kidnappedChancellor Palpatine, leader ofthe Galactic Senate.As the Separatist Droid Armyattempts to flee the besiegedcapital with their valuablehostage, two Jedi Knights lead adesperate mission to rescue thecaptive Chancellor....",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Revenge of the Sith",
                },
                "ref": null,
                "rendered": "Revenge of the Sith",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2005-05-19",
                },
                "ref": null,
                "rendered": "2005-05-19",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
